### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by Husain01**
-> Added Context to fetch videos and categories from API
-> Add Filter functionality according to the tab category chosen
-> Added Search Functionality
-> Added Single Video Page which is loaded with respect to the video ID.

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -16,6 +17,16 @@ export const Navbar = () => {
       type: "LOGOUT",
     });
   };
+  const [input, setInput] = useState("");
+  const { dispatch } = useData();
+  const searchHandler = (e) => {
+    if (e.key === "Enter" || e.keyCode === 8 || e.target.value === "") {
```
Comment (MarkVed17) [inline src/components/Navbar/Navbar.js:23]: I liked the way you handled events.

File: src/pages/Home/Home.js [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+import Aside from "../../components/Aside/Aside";
+import VideoCard from "../../components/VideoCard/VideoCard";
+import { useData } from "../../context/Video/VideoContext";
+import "./Home.css";
 
 export const Home = () => {
+  const { category, videos, dispatch, sortBy, search } = useData();
+
+  const sortHandler = (catName) => {
+    console.log(catName);
```
Comment (MarkVed17) [inline src/pages/Home/Home.js:11]: you may remove this.

**Review by MarkVed17 — APPROVED** [review]
Filters functioning as desired.
I really liked the UI.
You may proceed!

**Comment by HarshnaNHaswani** [comment]
- Clean, nicely formatted code
- Good folder structure
- seamless data flow
- Noticed the title of your app is still react app
- Loved the UI 
- (side-note: really liked the video choice)

**Review by HarshnaNHaswani — COMMENTED** [review]
Good to go

**Pull request merged**